File: domain/src/main/java/com/krillsson/sysapi/domain/health/HealthData.java
package com.krillsson.sysapi.domain.health;

public class HealthData {
    private final String description;
    private final double data;
    private final DataType dataType;

    public HealthData(String description, double data, DataType dataType) {
        this.description = description;
        this.data = data;
        this.dataType = dataType;
    }

    public String getDescription() {
        return description;
    }

    public double getData() {
        return data;
    }

    public DataType getDataType() {
        return dataType;
    }
}

File: sys-api-spring/src/main/kotlin/com/krillsson/sysapi/core/history/db/HealthData.kt
package com.krillsson.sysapi.core.history.db

import jakarta.persistence.Entity
import jakarta.persistence.FetchType
import jakarta.persistence.Id
import jakarta.persistence.JoinColumn
import jakarta.persistence.ManyToOne
import java.util.UUID

@Entity
class HealthData(
    @Id
    open var id: UUID,
    @JoinColumn(name = "historyId", insertable = false, updatable = false)
    @ManyToOne(fetch = FetchType.LAZY)
    open var history: HistorySystemLoadEntity? = null,
    open var historyId: UUID,
    open var description: String,
    open var data: Double,
    open var dataType: DataType
)